src/components/HideUserButton/BlockUserButton.tsx

```diff
@@ -0,0 +1,100 @@
+import { Button, ButtonProps, Menu } from '@mantine/core';
+import { openConfirmModal } from '@mantine/modals';
+import { IconUserCancel, IconUserCheck } from '@tabler/icons-react';
+import { MouseEventHandler } from 'react';
+import { LoginRedirect } from '~/components/LoginRedirect/LoginRedirect';
+import { useHiddenPreferencesData, useToggleHiddenPreferences } from '~/hooks/hidden-preferences';
+
+import { useCurrentUser } from '~/hooks/useCurrentUser';
+import { showSuccessNotification } from '~/utils/notifications';
+
+export function BlockUserButton({ userId, as = 'button', onToggleHide, ...props }: Props) {
+  const currentUser = useCurrentUser();
+
+  const users = useHiddenPreferencesData().blockedUsers;
+  const alreadyBlocking = users.some((x) => x.id === userId);
+  const toggleHiddenMutation = useToggleHiddenPreferences();
+
+  const handleBlockClick: MouseEventHandler<HTMLElement> = async (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    if (alreadyBlocking) {
+      toggleHiddenMutation
+        .mutateAsync({
+          kind: 'blockedUser',
+          data: [{ id: userId }],
+          hidden: !alreadyBlocking,
+        })
+        .then(() => {
+          showSuccessNotification({
+            title: alreadyBlocking ? 'User unblocked' : 'blocked',
+            message: `Content from this user will${
+              alreadyBlocking ? ' ' : ' not'
+            } show up in your feed`,
+          });
+        });
+    } else {
+      openConfirmModal({
+        title: 'Block User',
+        children: `Are you sure you want to block this user? Once a user is blocked, you won't see their content again and they won't see yours.`,
+        labels: { confirm: 'Yes, block the user', cancel: 'Cancel' },
+        confirmProps: { color: 'red' },
+        onConfirm: () =>
+          toggleHiddenMutation
+            .mutateAsync({
+              kind: 'blockedUser',
+              data: [{ id: userId }],
+              hidden: !alreadyBlocking,
+            })
+            .then(() => {
+              showSuccessNotification({
+                title: alreadyBlocking ? 'User unblocked' : 'blocked',
+                message: `Content from this user will${
+                  alreadyBlocking ? ' ' : ' not'
+                } show up in your feed`,
```

Comment:
Might wanna do this within the hook 🤷 ?

Reply:
You mean displaying the success notification? Since we are using that hook for different types of toggles, I feel this is the right place to do it for now. Otherwise, I'd need to take into account the other places we use it